llvm-general/src/LLVM/General/Internal/FFI/GlobalValueC.cpp

```diff
@@ -40,12 +40,36 @@ void LLVM_General_SetCOMDATSelectionKind(Comdat &comdat, unsigned csk) {
   comdat.setSelectionKind(Comdat::SelectionKind(csk));
 }
 
-LLVMBool LLVM_General_HasUnnamedAddr(LLVMValueRef globalVal) {
-	return unwrap<GlobalValue>(globalVal)->hasUnnamedAddr();
+typedef enum {
+    None   = 0,
+    Local  = 1,
+    Global = 2,
+} LLVM_General_UnnamedAddr;
```

Comment:
There's nothing to enforce the non-local correlation between this enum and the corresponding values in the the coding instances made by the call to genCodingInstance you've added to Module.hs. Please follow the example of how Atomicity is done to

- keep correctness as much as possible a consequence of locally verifiable correctness - e.g. lists of pairs where the elements of each pair correspond - as opposed to non-local correlations between files like this.
- minimize boilerplate through the use of metaprogramming - i.e. TH and ENUM_CASE / LLVM_GENERAL_FOR_EACH_*. Circuitous though the metaprogramming is, as much boilerplate as would be necessary without it quickly becomes unmanageable.